napari/_qt/qt_main_window.py

```diff
@@ -675,16 +675,13 @@ def __init__(self, viewer: 'Viewer', *, show: bool = True) -> None:
             self._update_theme_font_size
         )
 
-        self._add_viewer_dock_widget(
-            self._qt_viewer.dockConsole, tabify=False, menu=self.window_menu
-        )
+        self._add_viewer_dock_widget(self._qt_viewer.dockConsole, tabify=False)
         self._add_viewer_dock_widget(
             self._qt_viewer.dockLayerControls,
             tabify=False,
-            menu=self.window_menu,
         )
         self._add_viewer_dock_widget(
-            self._qt_viewer.dockLayerList, tabify=False, menu=self.window_menu
+            self._qt_viewer.dockLayerList, tabify=False
```

Comment:
Nice that we now remove this `menu` keyword now. I will note, this `menu` keyword is exposed publicly via `viewer.window.add_dock_widget`. I checked now using the `bbox_annotator.py` example and the toggling does work, but [looking at the code](https://github.com/napari/napari/blob/main/napari/_qt/qt_main_window.py#L1169) that takes care of it, we might want to update it to be more app-model-like, since right now it's creating a QAction directly by the looks of it? Anyway, not in scope for this PR

Reply:
Yes I saw this but did not realize it was public API. I think with app-model migration, this would not be used internally at all. I guess even if we offer another way for users to add items to the menu, this is a nice shortcut.

I'll make an issue for converting this to app-model.